### PR TITLE
chore: remove notifications experiment

### PR DESCRIPTION
**Pull request by dannykopping**
Notifications have proved stable in the [mainline release of v2.15](https://github.com/coder/coder/releases/tag/v2.15.0), and in preparation for v2.16 we're moving this to stable.

**Note:**

Assuming a config which relies on the defaults for notifications settings, customers will now start seeing logs like this (assuming they don't have an open SMTP server on localhost):

```
[API] 2024-09-30 09:44:44.695 [warn]  notifications.manager.notifier: message dispatch failed  notifier_id=4991afe2-66e2-4d89-835b-d737141cfec5  msg_id=eeaae4b1-70de-4e89-9d11-6f83ecb8bce2  method=smtp  attempt=1 ...
[API] error= SMTP client creation:
[API] github.com/coder/coder/v2/coderd/notifications/dispatch.(*SMTPHandler).dispatch.func1
[API] /Users/danny/Code/coder/coder/coderd/notifications/dispatch/smtp.go:128
[API] - establish plain connection to server:
[API] github.com/coder/coder/v2/coderd/notifications/dispatch.(*SMTPHandler).client
[API] /Users/danny/Code/coder/coder/coderd/notifications/dispatch/smtp.go:323
[API] - dial tcp [::1]:587: connect: connection refused
```

This occurs because the default notification method is set as `smtp` in `CODER_NOTIFICATIONS_METHOD`, and other defaults are configured (`CODER_NOTIFICATIONS_EMAIL_SMARTHOST` and `CODER_NOTIFICATIONS_EMAIL_HELLO`).

This is not harmful, but it is _avoidable_. We'd have to remove the default values for `CODER_NOTIFICATIONS_EMAIL_SMARTHOST` and `CODER_NOTIFICATIONS_EMAIL_HELLO`, and detect their absence, and log a warning with a link to the docs.

Effectively it amounts to the same thing, but it's a little more user-friendly.

The challenge with this approach, though, is it might remove some actually useful default values for customers. We could cover this in release notes, but it weakens our backwards-compatibility posture. It's not strictly a break, but it could cause unintended behaviour if operators blindly upgrade without reading the notes (everybody reads the release notes, right? RIGHT?) _and_ they rely on these defaults.

I think a good middleground could be to mention the docs in the above error log. Then, when we implement https://github.com/coder/coder/issues/14644, we remove the default SMTP values and set a `none` method by default.

---

In retrospect I should've tried to find a way to not define any defaults here, but we live and learn.

### CODE REVIEW

File: coderd/notifications_test.go [hunk 1]
```diff
@@ -20,7 +20,6 @@ func createOpts(t *testing.T) *coderdtest.Options {
 	t.Helper()
 
 	dt := coderdtest.DeploymentValues(t)
-	dt.Experiments = []string{string(codersdk.ExperimentNotifications)}
```
Comment (mtojek) [inline coderd/notifications_test.go:23]: nit: maybe use comment instead or empty array with `FIXME`?

File: cli/server.go [hunk 2]
```diff
@@ -56,15 +56,16 @@ import (
...
 	"github.com/coder/pretty"
 	"github.com/coder/quartz"
 	"github.com/coder/retry"
 	"github.com/coder/serpent"
 	"github.com/coder/wgtunnel/tunnelsdk"
 
+	"github.com/coder/coder/v2/coderd/entitlements"
+	"github.com/coder/coder/v2/coderd/notifications/reports"
+	"github.com/coder/coder/v2/coderd/runtimeconfig"
+
```
Comment (mtojek) [inline cli/server.go:68]: nit: without empty line?

File: coderd/notifications_test.go (hunk 1 above)
Comment (dannykopping) [reply coderd/notifications_test.go:23]: With the feature not being experimental anymore and `Experiments` already having a zero value of an empty slice, I think removing this is appropriate.

File: cli/server.go (hunk 2 above)
Comment (dannykopping) [reply cli/server.go:68]: 🤷 this is just what `make fmt` did for me.

**Comment by stirby** [comment]
@dannykopping thanks for calling out that note. We should ship as is and revisit this before the next release.

I'd opt to keep the defaults with a note in the docs rather than introduce friction in backward compatibility. To improve visibility, I can link to this PR in the release notes for 2.16.0. 

We get a lot of negative feedback on the difficulty of reverting upgrades, and it'd be unfortunate for a downgrade to override their settings.


Can we merge this ASAP so I can pull it into the release branch? Just whenever you all are online. (cc @mtojek)

File: coderd/notifications/reports/generator.go [hunk 3]
```diff
@@ -49,7 +49,7 @@ func NewReportGenerator(ctx context.Context, logger slog.Logger, db database.Sto
 				return nil
 			}
 
-			err = reportFailedWorkspaceBuilds(ctx, logger, db, enqueuer, clk)
+			err = reportFailedWorkspaceBuilds(ctx, logger, tx, enqueuer, clk)
```
Comment (dannykopping) [inline coderd/notifications/reports/generator.go:52]: Passing the main `db` object through caused a deadlock; we need to pass the transaction-scoped db object instead.

**Review by mtojek — APPROVED** [review]

**Comment by dannykopping** [comment]
I've added the beta badge:

image

image

File: site/src/pages/ManagementSettingsPage/SidebarView.tsx [hunk 4]
```diff
@@ -148,11 +148,12 @@ const DeploymentSettingsNavigation: FC<DeploymentSettingsNavigationProps> = ({
 							Users
 						</SidebarNavSubItem>
 					)}
-					{experiments.includes("notifications") && (
+					<Stack direction={"row"} alignItems={"center"} css={{ gap: 0 }}>
```
Comment (BrunoQuaresma) [inline site/src/pages/ManagementSettingsPage/SidebarView.tsx:151]: Nit: When passing string values directly you can pass the value directly without the brackets:
```
direction="row"
```

**Comment by BrunoQuaresma** [comment]
I tried to QA this locally, but I don't see the option in the sidebar when starting the server without any experiments set. Is it expected? I'm not sure how we are handling beta features tho.

**Review by BrunoQuaresma — APPROVED** [review]
UI and code look good! 👍

**Pull request merged**